### DISCUSSION TITLE
SP Modern Search not finding/returning all refinable properties

### DISCUSSION THREAD

**Discussion by 0204xls** [microsoft-search/pnp-modern-search#3088] [Q&A]
Dear Community,

I am running into a strange problem for which I cannot find a solution. Would appreciate any input or suggestions. Problem is as follows:

Modern search does not find or register _some_ items and metadata-- seemingly at random. For example, when I set the vertical in my search page to display one library, it will return 22 documents when the library actually contains 33. Of those 22 documents, some are missing metadata. For instance, Document A will have a value for the site column called 'DocumentType' but Document B (in the same library) will not show any value. This means I cannot correctly filter the results. 

If I look at debug view, Document B's value is 'null' in the relevant 'RefinableStringXX' field. If I look at the document in the SP Online library though, the item has a value in that column and all metadata is filled out like normal. Any idea why the search is not picking up only these certain documents/metadata? It happens in all libraries on the site.

Steps for re-creating problem:
1. Create libraries in site.
2. Create custom site columns.
3. Populate libraries with documents and accompanying metadata in the custom site columns.
4. Map respective crawled properties from custom site columns to Refinable properties.
5. Wait 48+ hours for reindexing. 
6. Set up Modern Search page in the site with verticals and filters. Select all relevant properties, connect the webparts, etc.
7. Enable debug mode and look through any search results.
8. RefinableString/Date properties for many items are 'null'.
9. Not all items from libraries appear in search results.

I can also provide more information if needed. Thank you in advance!

**Comment by kasperbolarsen** [comment]
This question is not really about the PnP Modern Search web parts but about basic SharePoint Search. This kind of questions should be asked in this forum: https://techcommunity.microsoft.com/t5/sharepoint/bd-p/SharePoint_General
